<launch>
    <!-- PERCEPTION -->
    <node name="image_view_hsv" pkg="image_view" type="image_view" respawn="false" output="log">
        <remap from="image" to="/hsv_detector/image_result"/>
        <param name="autosize" value="true" />
    </node>

    <!-- <node name="image_view_hsv2" pkg="image_view" type="image_view" respawn="false" output="log">
        <remap from="image" to="/hsv_detector/image_result_thres"/>
        <param name="autosize" value="true" />
    </node> -->

    <include file="$(find baxter_collaboration)/launch/baxter_marker_publisher.launch" />

    <param name="hsv_detector/reference_frame" value="/base"/>
    <param name="hsv_detector/camera_frame"    value="/right_hand_camera"/>
    <param name="hsv_detector/area_threshold"  value="250"/>

    <node pkg="baxter_collaboration" type="hsv_detector" name="hsv_detector" output="screen" respawn="true">
        <remap from="hsv_detector/image" to="/cameras/right_hand_camera/image"/>
        <remap from="hsv_detector/camera_info" to="/cameras/right_hand_camera/camera_info"/>
    </node>

    <!-- Objects database for the left arm -->
    <rosparam param = "action_provider/objects_left">
        "table_top"   : 200
        "leg_1"       : 150
        "leg_2"       : 151
        "leg_3"       : 152
        "leg_4"       : 153
    </rosparam>

    <!-- Objects database for the right arm -->
    <rosparam param = "action_provider/objects_right">
        "screwdriver"  : 1
        "screws_box"   : 2
        "brackets_box" : 3
    </rosparam>

    <!-- Objects database for the hsv cartesian detector -->
    <rosparam param = "hsv_detector/objects_db">
        "screwdriver" : { size: [0.099, 0.030], HSV: { H: [160,  10], S: [ 70, 166], V: [10,  66]}}
         <!-- This should be the blue box -->
        "screws_box"  : { size: [0.111, 0.111], HSV: { H: [ 80, 130], S: [120, 216], V: [70, 146]}}
         <!-- This should be the green box -->
        "brackets_box": { size: [0.111, 0.111], HSV: { H: [ 30, 100], S: [ 80, 246], V: [20,  96]}}
    </rosparam>

    <!-- CONTROL -->
    <!-- If the system should use the robot or not (useful for debugging) -->
    <arg name="useRobot"            default="true"/>
    <!-- If the system should use the robot or not (useful for debugging) -->
    <param name="action_provider/use_robot"             value="$(arg useRobot)"/>

    <!-- Parameters for force interaction -->
    <arg name="forceFilterVarianceL" default="0.001"/>
    <arg name="forceFilterVarianceR" default="0.01"/>
    <arg name="relativeForceThresholdL"   default="65.0"/>
    <arg name="relativeForceThresholdR"   default="500.0"/>

    <param name="action_provider/force_filter_variance_left"        value="$(arg forceFilterVarianceL)"/>
    <param name="action_provider/force_filter_variance_right"       value="$(arg forceFilterVarianceR)"/>
    <param name="action_provider/relative_force_threshold_left"     value="$(arg relativeForceThresholdL)"/>
    <param name="action_provider/relative_force_threshold_right"    value="$(arg relativeForceThresholdR)"/>

    <!-- If the robot will recover from an error or will wait for the external planner's decisions -->
    <arg name="internalRecovery"    default="true"/>
    <param name="action_provider/internal_recovery"     value="$(arg internalRecovery)"/>

    <node pkg="baxter_collaboration" type="modular_action_provider" name="modular_action_provider" output="screen"/>

    <!-- VISUALIZATION & MISC -->
    <!-- Parameter for speech pop up duration -->
    <param name="baxter_display/speech_duration"        value="16.0"/>

    <node pkg="baxter_collaboration" type="baxter_display"  name="baxter_display"  output="screen"/>

    <include file="$(find svox_tts)/launch/svox_tts.launch" />

    <include file="$(find ros_speech2text)/launch/ros_speech2text.launch"/>
</launch>

